frontend/src/components/teamsAndOrgs/teams.js

```diff
@@ -163,38 +172,56 @@ export function TeamInformation(props) {
-          <FormattedMessage {...messages.inviteOnly} />
+          <FormattedMessage {...messages.joinMethod} />
         </label>
-        <Field name="inviteOnly" className={fieldClasses}>
-          {(props) => (
-            <div className="fl">
-              <SwitchToggle
-                isChecked={props.input.value}
-                onChange={props.input.onChange}
-                label={<FormattedMessage {...messages.inviteOnlyDescription} />}
-                labelPosition="right"
-              />
-            </div>
-          )}
-        </Field>
+        {joinMethods.map((method) => (
+          <div className="pv2">
+            <RadioField name="joinMethod" value={method.value} />
+            <span className="f5">
+              <FormattedMessage {...messages[method.descriptionId]} />
+            </span>
+            <InfoIcon
+              width={12}
+              height={12}
+              className="blue-grey v-mid pb1 ml2"
+              data-tip={intl.formatMessage(messages[`${method.descriptionId}Description`])}
+            />
+            <ReactTooltip place="bottom" className="mw6" effect="solid" />
+          </div>
+        ))}
       </div>
-      <div className="cf pt1">
-        <label className={labelClasses}>
-          <FormattedMessage {...messages.visibility} />
-        </label>
-        <div className="pv2">
-          <RadioField name="visibility" value="PUBLIC" />
-          <span className="fw8 f5">
-            <FormattedMessage {...messages.public} />
-          </span>
-        </div>
-        <div className="pv2">
-          <RadioField name="visibility" value="PRIVATE" />
-          <span className="fw8 f5">
-            <FormattedMessage {...messages.private} />
-          </span>
+      {formState.values.joinMethod === 'BY_INVITE' ? (
+        <div className="cf pt1">
+          <label className={labelClasses}>
+            <FormattedMessage {...messages.visibility} />
+          </label>
+          <div className="pv2">
+            <RadioField name="visibility" value="PUBLIC" />
+            <span className=" f5">
+              <FormattedMessage {...messages.public} />
+            </span>
+            <InfoIcon
+              width={12}
+              height={12}
+              className="blue-grey v-mid pb1 ml2"
+              data-tip={intl.formatMessage(messages["publicDescription"])}
+            />
+          </div>
+          <div className="pv2">
+            <RadioField name="visibility" value="PRIVATE" />
+            <span className="f5">
+              <FormattedMessage {...messages.private} />
+            </span>
+            <InfoIcon
+              width={12}
+              height={12}
+              className="blue-grey v-mid pb1 ml2"
+              data-tip={intl.formatMessage(messages["privateDescription"])}
+            />
+            <ReactTooltip place="bottom" className="mw6" effect="solid" />
+          </div>
         </div>
-      </div>
+      ) : null}
```

Comment:
Was there any reason to utilize a ternary operator instead of the logical && operator? For the conditional rendering part, I would have used the logical && rather than the ternary operator because the underlying component already renders something from the code above.

I assume we could also map these two visibility values from an array, similar to how you did with the join methods above this code.